src/header/common/mod.rs

```diff
@@ -76,6 +76,51 @@ macro_rules! deref(
     }
 );
 
+macro_rules! impl_list_header(
+    ($from:ident, $name:expr, $item:ty) => {
+        deref!($from => $item);
+
+        impl header::Header for $from {
+            fn header_name(_: Option<$from>) -> &'static str {
+                $name
+            }
+
+            fn parse_header(raw: &[Vec<u8>]) -> Option<$from> {
+                shared::from_comma_delimited(raw).map($from)
+            }
+        }
+
+        impl header::HeaderFormat for $from {
+            fn fmt_header(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+                shared::fmt_comma_delimited(fmt, &self[])
+            }
+        }
+    }
+);
+
+macro_rules! impl_header(
+    ($from:ident, $name:expr, $item:ty) => {
+        deref!($from => $item);
+
+        impl header::Header for $from {
+            fn header_name(_: Option<$from>) -> &'static str {
+                $name
+            }
+
+            fn parse_header(raw: &[Vec<u8>]) -> Option<$from> {
+                from_one_raw_str(raw).map($from)
+            }
+        }
+
+        impl header::HeaderFormat for $from {
+            fn fmt_header(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+                let value = &(self[]);
+                value.fmt(fmt)
```

Comment:
Now that Show and String are common, this could call the wrong one depending on which trait is in scope. Try explicitly using `fmt::String::fmt(&value, fmt)`.